main.go

```diff
@@ -247,6 +247,5 @@ func main() {
 
 	go p.Serve()
 
-	for range time.Tick(time.Second) {
-	}
+	runtime.Goexit()
```

Comment:
some googling suggests that 

```
In general, you should return normally from your goroutines to keep the code clear, and prevent unexpected behavior.
```

https://stackoverflow.com/questions/31657129/return-vs-goexit-for-goroutines